toolsrc/src/vcpkg-test/format-manifest.cpp

```diff
@@ -0,0 +1,127 @@
+        ports / fs::u8path("manifest_port" SEPARATOR "vcpkg.json"),
+        ports / fs::u8path("ambiguous_port"),
+        ports / fs::u8path("ambiguous_port" SEPARATOR "CONTROL"),
+        ports / fs::u8path("ambiguous_port" SEPARATOR "vcpkg.json"),
+        // conflict between port name and filesystem name test cases
+        original_cwd / fs::u8path("overlap_port"),
+        ports / fs::u8path("overlap_port"),
+        ports / fs::u8path("overlap_port" SEPARATOR "CONTROL"),
+    }};
+}
+
+TEST_CASE ("resolves_existing_absolute_path", "[commands][format-manifest]")
+{
+    const auto result = resolve_format_manifest_input(existing_absolute, original_cwd, ports, filesystem);
+    CHECK(result.has_value());
+    CHECK(result.value_or_exit(VCPKG_LINE_INFO) == existing_absolute_path);
+}
+
+TEST_CASE ("does_not_resolve_missing_absolute_path", "[commands][format-manifest]")
+{
+    const auto result = resolve_format_manifest_input(missing_absolute, original_cwd, ports, filesystem);
+    CHECK(!result.has_value());
+    CHECK(result.error().find(" not found.") != std::string::npos);
+}
+
+TEST_CASE ("resolves_relative_paths", "[commands][format-manifest]")
+{
+    vcpkg::StringView relative_paths[] = {
+        vcpkg::StringView{"example"},
+        vcpkg::StringView{"example" SEPARATOR "CONTROL"},
+        vcpkg::StringView{"example" SEPARATOR "vcpkg.json"},
+        vcpkg::StringView{"example" SEPARATOR "anything.json"},
+    };
+
+    for (vcpkg::StringView& relative : relative_paths)
+    {
+        const auto result = resolve_format_manifest_input(relative, original_cwd, ports, filesystem);
+        CHECK(result.has_value());
+        CHECK(result.value_or_exit(VCPKG_LINE_INFO) == original_cwd / fs::u8path(relative));
+    }
+}
+
+TEST_CASE ("resolves_control_port", "[commands][format-manifest]")
+{
+    const auto result = resolve_format_manifest_input("control_port", original_cwd, ports, filesystem);
+    CHECK(result.has_value());
+    CHECK(result.value_or_exit(VCPKG_LINE_INFO) == ports / fs::u8path("control_port" SEPARATOR "CONTROL"));
+}
+
+TEST_CASE ("resolves_manifest_port", "[commands][format-manifest]")
+{
+    const auto result = resolve_format_manifest_input("manifest_port", original_cwd, ports, filesystem);
+    CHECK(result.has_value());
+    CHECK(result.value_or_exit(VCPKG_LINE_INFO) == ports / fs::u8path("manifest_port" SEPARATOR "vcpkg.json"));
+}
+
+TEST_CASE ("does_not_resolve_ambiguous_port", "[commands][format-manifest]")
+{
+    const auto result = resolve_format_manifest_input("ambiguous_port", original_cwd, ports, filesystem);
+    CHECK(!result.has_value());
+    CHECK(result.error().find("Both a manifest file and a CONTROL file exist") != std::string::npos);
+}
+
+TEST_CASE ("chooses_filesystem_path_over_port_name", "[commands][format-manifest]")
+{
+    const auto result = resolve_format_manifest_input("overlap_port", original_cwd, ports, filesystem);
+    CHECK(result.has_value());
```

Comment:
```suggestion
    REQUIRE(result.has_value());
```
`REQUIRE` will bail the test case immediately, preventing a crash on the line below; `CHECK()` will continue.

https://github.com/catchorg/Catch2/blob/devel/docs/assertions.md#natural-expressions